src/features/goblins/storageHouse/components/DeliverItems.tsx

```diff
@@ -192,104 +169,102 @@ export const DeliverItems: React.FC<Props> = ({ onWithdraw }) => {
-        <div className="flex flex-wrap h-fit -ml-1.5 mb-2">
-          {inventoryItems.map((itemName) => {
-            const details = ITEM_DETAILS[itemName];
-
-            const totalCountOfItemType = inventory[itemName] || new Decimal(0);
-
-            return (
+    <>
+      <div className="p-2 mb-2">
+        <div className="mt-3">
+          <h2 className="mb-1 text-sm">Inventory:</h2>
+          <div className="flex flex-wrap h-fit -ml-1.5 mb-2">
+            {(Object.keys(inventory) as InventoryItemName[]).map((itemName) => (
               <Box
-                count={totalCountOfItemType}
                 key={itemName}
+                count={inventory[itemName]}
                 onClick={() => onAdd(itemName)}
-                image={details.image}
-                canBeLongPressed
+                image={ITEM_DETAILS[itemName].image}
               />
-            );
-          })}
-        </div>
+            ))}
+          </div>
 
-        <div className="mt-2 min-h-[64px]">
-          <h2 className="text-sm">You will receive:</h2>
-          <div className="mt-2 -ml-1.5">
-            {selectedItems
-              .sort((a, b) => KNOWN_IDS[a] - KNOWN_IDS[b])
-              .map((itemName) => {
-                return (
-                  <div className="flex items-center pl-1" key={itemName}>
-                    <div className="w-80 flex items-center">
-                      <Box
-                        hideCount
-                        count={selected[itemName]}
-                        key={itemName}
-                        onClick={() => onSubtract(itemName)}
-                        image={ITEM_DETAILS[itemName].image}
-                        canBeLongPressed
-                      />
-                      <div className="flex flex-col">
-                        <span className="text-sm">{`${parseFloat(
-                          selected[itemName]
-                            ?.mul(1 - DELIVERY_FEE / 100)
-                            .toFixed(4, Decimal.ROUND_DOWN) as string
-                        )} ${itemName}`}</span>
-                        <div className="flex">
-                          <span className="text-xxs">{`${parseFloat(
-                            selected[itemName]
-                              ?.mul(DELIVERY_FEE / 100)
-                              .toFixed(4, Decimal.ROUND_DOWN) as string
-                          )} Goblin fee`}</span>
-                          <img src={goblinHead} className="w-6 ml-2" />
-                        </div>
-                      </div>
-                    </div>
-
-                    <img
-                      src={SUNNYSIDE.icons.cancel}
-                      className="h-4 cursor-pointer"
-                      onClick={() => onRemove(itemName)}
+          <div className="mt-2 min-h-[64px]">
+            <h2 className="text-sm">Items to deliver:</h2>
+            {(Object.keys(selected) as InventoryItemName[]).map((itemName) => (
```

Comment:
You should be able to use `getKeys` here to hold the typing.

Reply:
That's a handy function, thank you, fixed!